contracts/SpokePool.sol

```diff
@@ -323,6 +401,32 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         }
     }
 
+    struct FillRelayEventData {
+        bytes32 relayHash;
+        RelayData relayData;
+        uint64 updatableRelayerFeePct;
+        uint256 fillAmountPreFees;
+        uint256 repaymentChain;
+    }
+
+    function _emitFillRelay(FillRelayEventData memory eventData) internal {
```

Comment:
Is this extra method here just to avoid stack too deep errors?

Reply:
exactly